File: go/lc102.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func levelOrder(root *TreeNode) [][]int {
	res, cur := make([][]int, 0), make([]*TreeNode, 0)
	if root == nil {
		return res
	}
	cur = append(cur, root)
	for len(cur) > 0 {
		next := make([]*TreeNode, 0)
		temp := make([]int, 0)
		for _, now := range cur {
			temp = append(temp, now.Val)
			if now.Left != nil {
				next = append(next, now.Left)
			}
			if now.Right != nil {
				next = append(next, now.Right)
			}
		}
		cur = next
		res = append(res, temp)
	}
	return res
}

File: java/com/lc/lc102.java
package lc;

import java.util.*;

public class lc102 {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> res = new ArrayList<>();
        if (root == null) {
            return res;
        }

        LinkedList<TreeNode> q = new LinkedList<>();
        q.addLast(root);
        while (!q.isEmpty()){
            List<Integer> tmp = new ArrayList<>();
            int cnt = q.size();
            for (int i = 0; i < cnt; i++){
                TreeNode node = q.poll();
                if (node.left != null){
                    q.addLast(node.left);
                }
                if (node.right != null){
                    q.addLast(node.right);
                }
                tmp.add(node.val);
            }
            res.add(tmp);
        }

        return res;
    }
}
